client/my-sites/github-deployments/components/deployments-survey/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { translate } from 'i18n-calypso';
+import ReactDOM from 'react-dom';
+import surveyImage from 'calypso/assets/images/illustrations/github-deployments.svg';
+import SurveyModal from 'calypso/components/survey-modal';
+import { maybeSetDeploymentDone, shouldShowDeploymentSurvey } from './utils';
+
+export const GitHubDeploymentSurvey = () => {
+	const localeSlug = useLocale();
+
+	maybeSetDeploymentDone();
```

Comment:
Let's not run side effects on render. Let's wrap it with `useEffect` or `useLayoutEffect` if it needs to be run before painting.